src/librustc_resolve/diagnostics.rs

```diff
@@ -197,6 +197,66 @@ See the Types section of the reference for more information about the primitive
 types:
 
 http://doc.rust-lang.org/reference.html#types
+"##,
+
+E0364: r##"
+Private items cannot be publicly re-exported.  This error indicates
+that you attempted to write a 'pub use' against a type or value that was
+not itself public.
+
+Here is an example that demonstrates the error:
+
+```
+mod foo {
+    const X: u32 = 1;
+}
+pub use foo::X;
+```
+
+The solution to this problem is to ensure that the items that you are
+re-exporting are themselves marked with 'pub':
+
+```
+mod foo {
+    pub const X: u32 = 1;
+}
+pub use foo::X;
+```
+
+See the Use Declarations section of the reference for more information
```

Comment:
FWIW using [1] for footnoting/citing is common from the pre-markdown days (though you generally don't put brackets around the text, so it would be `Use Declarations[1]`). I think it's okay to do a link like this.